source/MaterialXRuntime/Commands/MetadataCommands.cpp

```diff
@@ -0,0 +1,50 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+#include <MaterialXRuntime/Commands/MetadataCommands.h>
+
+#include <MaterialXRuntime/Private/Commands/PvtSetMetadataCmd.h>
+#include <MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.h>
+
+#include <MaterialXRuntime/Private/PvtApi.h>
+
+#include <MaterialXRuntime/RtTypeDef.h>
+
+namespace MaterialX
+{
+
+namespace RtCommand
+{
+
+void setMetadata(const RtObject& src, const RtToken& metadataName, const string& metadataValue, RtCommandResult& result)
+{
+    try
+    {
+        RtValue v = RtValue::createNew(RtType::STRING, src);
```

Comment:
I realize we are creating a new value instance here for every call. For large value types, like string, this means we are allocating this in the corresponding prim's allocator, which is not deallocated until the prim is destroyed. So it's a type of memory leak.

We do have the same issue in the setAttribute commands, so it's not just here. I think we need to fix this at some point. I will log an issue for it. Just wanted to mention it now that I discovered this.